storage/src/test/java/org/apache/kafka/server/log/remote/metadata/storage/ConsumerTaskTest.java

```diff
@@ -84,42 +78,41 @@ public class ConsumerTaskTest {
 
     private ConsumerTask consumerTask;
     private MockConsumer<byte[], byte[]> consumer;
-    private Thread thread;
 
     @BeforeEach
     public void beforeEach() {
         final Map<TopicPartition, Long> offsets = remoteLogPartitions.stream()
             .collect(Collectors.toMap(Function.identity(), e -> 0L));
-        consumer = spy(new MockConsumer<>(OffsetResetStrategy.EARLIEST));
+        consumer = new MockConsumer<>(OffsetResetStrategy.EARLIEST);
         consumer.updateBeginningOffsets(offsets);
         consumerTask = new ConsumerTask(handler, partitioner, consumer, 10L, 300_000L, Time.SYSTEM);
-        thread = new Thread(consumerTask);
     }
 
     @AfterEach
-    public void afterEach() throws InterruptedException {
-        if (thread != null) {
-            assertDoesNotThrow(() -> consumerTask.close(), "Close method threw exception");
-            thread.join(10_000);
-            assertFalse(thread.isAlive(), "Consumer task thread is still alive");
-        }
+    public void afterEach() {
+        assertDoesNotThrow(() -> consumerTask.close(), "Close method threw exception");
+        assertDoesNotThrow(() -> consumerTask.closeConsumer(), "CloseConsumer method threw exception");
+        assertTrue(consumer.closed());
     }
 
     /**
      * Tests that the consumer task shuts down gracefully when there were no assignments.
      */
     @Test
-    public void testCloseOnNoAssignment() throws InterruptedException {
-        thread.start();
-        Thread.sleep(10);
+    public void testCloseOnNoAssignment() {
         assertDoesNotThrow(() -> consumerTask.close(), "Close method threw exception");
+        assertDoesNotThrow(() -> consumerTask.closeConsumer(), "CloseConsumer method threw exception");
     }
 
     @Test
     public void testIdempotentClose() {
-        thread.start();
+        // Go through the closure process
         consumerTask.close();
+        consumerTask.closeConsumer();
+
+        // Go through the closure process again
         consumerTask.close();
+        consumerTask.closeConsumer();
```

Comment:
nit: closeConsumer is normally only called once during thread exit, so it's probably not a good idea to call it again here.